### ISSUE TITLE
HTTP-based metricbeat modules can no longer perform authentication

### ISSUE THREAD

**Issue by ycombinator** [elastic/beats#11351]
- Version: `master` / `8.0.0`
- Operating System: Mac OS X
- Steps to Reproduce: 
  1. Setup Elasticsearch with Trial license, Security enabled, and `elastic` user to have password as `abc123`.
  1. Enable Elasticsearch metricbeat module.
     ```
     ./metricbeat modules enable elasticsearch
     ```
  2. Uncomment and set `username` and `password` settings in the Elasticsearch metricbeat module configuration.
     ```
     username: elastic
     password: abc123
     ```
  3. Start Metricbeat.
     ```
     ./metricbeat -e
     ```

You will see a 401 Unauthorized error in the logs. 

Looking at Wireshark, it looks like Metricbeat does not add the `Authorization` header, as one would expect it to with the above configuration:

<img width="1547" alt="Screen Shot 2019-03-20 at 12 43 02 PM" src="https://user-images.githubusercontent.com/51061/54714246-b8499980-4b0d-11e9-8480-dc0bbf725aa4.png">

I did a similar test with the Kibana Metricbeat module and ended up with the same results: 401 error in the logs and Wireshark showing no `Authorization` header.

**Comment by ycombinator** [comment]
Just tested with 6.7.0 and this was working as expected.

**Comment by ycombinator** [comment]
Just tested with Metricbeat 7.0.0 and this is working as expected.

**Comment by ycombinator** [comment]
Just tested with Metricbeat 7.1.0 (built from the `7.x` branch) and this is working as expected.

**Comment by ycombinator** [comment]
Based on the above versions' testing, it appears that this regression was introduced in `master` / `8.0.0` recently, that is between the last time `master` was backported to `7.x` and now.

@ruflin @jsoriano can you think of any recent PRs to `master` that might've introduced this regression?

**Comment by ycombinator** [comment]
@jsoriano @ruflin I did some git bisect testing and I think https://github.com/elastic/beats/pull/11032 might've caused this regression. Would one of you mind confirming?

**Comment by jsoriano** [comment]
I'd also bet this regresion is introduced by #11032, investigating it.

@ycombinator could you try to reproduce this issue with 6.7.0 but setting a timeout? So the config is something like:

```
username: elastic
password: abc123
timeout: 2s
```

**Comment by jsoriano** [comment]
@ycombinator PR open to fix this #11353, good catch!
